File: backend/BudgetToWealthApi/Models/InvestmentReturns/ManualInvestmentReturn.cs
public class ManualInvestmentReturn : BaseEntity
{
    public required Guid ManualInvestmentCategoryId { get; set; }
    public ManualInvestmentCategory? ManualInvestmentCategory { get; set; }
    public required DateOnly StartDate { get; set; }
    public required DateOnly EndDate { get; set; }
    public required decimal ManualInvestmentPercentageReturn { get; set; }
    public RecurrenceFrequency? ManualInvestmentRecurrenceFrequency { get; set; }
    public DateOnly? ManualInvestmentRecurrenceEndDate { get; set; }
    public string? UserId { get; set; }
}

File: frontend/app/net-worth/investment-returns/components/ManualInvestmentReturn.ts
import { RecurrenceFrequency } from "@/app/cashflow";
import { ListTableItem, Category } from "@/app/components";

export type ManualInvestmentReturn = ListTableItem & {
  id?: number;
  manualInvestmentCategoryId: string;
  manualInvestmentCategory?: Category;
  startDate: string;
  endDate: string;
  manualInvestmentPercentageReturn: number;
  manualInvestmentRecurrenceFrequency?: RecurrenceFrequency;
  manualInvestmentRecurrenceEndDate?: string;
  userId?: string;
  name?: string;
}


export const getManualInvestmentReturnDisplayName = (investmentReturn: ManualInvestmentReturn) => {
  return `${investmentReturn.manualInvestmentCategory?.name}`;
}
